dsl/src/main/kotlin/com/chutneytesting/kotlin/synchronize/ChutneyServerServiceImpl.kt

```diff
@@ -34,8 +33,19 @@ object ChutneyServerServiceImpl : ChutneyServerService {
         return HttpClient.get(serverInfo, "/api/scenario/component-edition/$scenarioId")
     }
 
-    override fun updateJsonScenario(serverInfo: ChutneyServerInfo, scenario: ChutneyScenario) {
-        val remoteScenario: LinkedHashMap<String, Any> = getRemoteScenario(serverInfo, scenario.id!!)
+    override fun createOrUpdateJsonScenario(serverInfo: ChutneyServerInfo, scenario: ChutneyScenario): Int {
+        if (scenario.id != null) {
```

Comment:
I prefer to have some thing more explicit:
if(new scenario) {
create it
} else {
update it
}